internal/kms/aws_metadata.go

```diff
@@ -125,7 +125,12 @@ func initAWSMetadataKMS(args ProviderInitArgs) (EncryptionKMS, error) {
 }
 
 func (kms *AWSMetadataKMS) getSecrets() (map[string]interface{}, error) {
-	c := k8s.NewK8sClient()
+	c, err := k8s.NewK8sClient()
+	if err != nil {
+		return nil, fmt.Errorf("can not connect to Kubernetes to get "+
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to connect to Kubernetes to get "+
```
:sweat_smile:  @nixpanic  Any reason to not follow this style as we have already discussed ?

Reply:
Not sure if there is something specific we settled on? I can surely change it, but we should consider having some checks in the CI for this (is there an error-format-checker?).

If you want me to change this, just mark it **Request changes** so the PR is blocked from  merging :-)

Reply:
I dont mind either way, was just wondering if there was a reason for `can not get` wording here.
Yes I think we need some ci checks 🤔.
https://github.com/ceph/ceph-csi/issues/2371#issuecomment-894085816

Reply:
I think we need to follow the format `failed to ..context.. : %v` so that it helps in debugging too(to know exactly the flow of the error)

Reply:
some errors have a prefix like `can not get ConfigMap ..., failed to connect ...`. I think that is acceptable :smile: